src/Microsoft.Diagnostics.ExtensionCommands/DumpExceptionsCommand.cs

```diff
@@ -0,0 +1,122 @@
+    public class DumpExceptionsCommand : CommandBase
+    {
+        [ServiceImport]
+        public ClrRuntime Runtime { get; set; } = null!;
+
+        [ServiceImport]
+        public LiveObjectService LiveObjects { get; set; } = null!;
+
+        [Option(Name = "-live")]
+        public bool Live { get; set; }
+
+        [Option(Name = "-dead")]
+        public bool Dead{ get; set; }
+
+        [Option(Name = "-gen")]
+        public string? Generation { get; set; }
+
+        [Option(Name = "-type")]
+        public string? Type { get; set; }
+
+        public override void Invoke()
+        {
+            HeapWithFilters filteredHeap = ParseArguments();
+
+            IEnumerable<ClrObject> exceptionObjects =
+                filteredHeap.EnumerateFilteredObjects(Console.CancellationToken)
+                    .Where(obj => obj.IsException);
+
+            if (Live)
+            {
+                exceptionObjects = exceptionObjects.Where(obj => LiveObjects.IsLive(obj));
+            }
+
+            if (Dead)
+            {
+                exceptionObjects = exceptionObjects.Where(obj => !LiveObjects.IsLive(obj));
+            }
+
+            if (!string.IsNullOrWhiteSpace(Type))
+            {
+                string type = Type!;
+                exceptionObjects = exceptionObjects.Where(obj => obj.Type!.Name!.IndexOf(type, StringComparison.OrdinalIgnoreCase) >= 0);
+            }
+
+            PrintExceptions(exceptionObjects);
+        }
+
+        private void PrintExceptions(IEnumerable<ClrObject> exceptionObjects)
+        {
+            Table output = new(Console, ColumnKind.Pointer, ColumnKind.Pointer, ColumnKind.TypeName);
+            output.WriteHeader("Address", "MethodTable", "Message", "Name");
+
+            int totalExceptions = 0;
+            foreach (ClrObject exceptionObject in exceptionObjects)
+            {
+                totalExceptions++;
+
+                ClrException clrException = exceptionObject.AsException()!;
+                output.WriteRow(exceptionObject.Address, exceptionObject.Type!.MethodTable, exceptionObject.Type!.Name);
+
+                Console.Write("        Message: ");
+                Console.WriteLine(clrException.Message ?? "<null>");
+
+                ImmutableArray<ClrStackFrame> stackTrace = clrException.StackTrace;
+                if (stackTrace.Length > 0)
+                {
+                    Console.Write("        StackFrame: ");
+                    Console.WriteLine(stackTrace[0].ToString());
+                }
+            }
+
+            Console.WriteLine();
+            Console.WriteLine($"    Total: {totalExceptions} objects");
+        }
+
+        private HeapWithFilters ParseArguments()
+        {
+            HeapWithFilters filteredHeap = new(Runtime.Heap);
+
+            if (Live && Dead)
+            {
+                Live = false;
+                Dead = false;
+            }
+
+            // TODO: CR: this is the same as in dumpheap. Where to put this?
```

Comment:
You can leave it for now until we figure out the best place for it.

Reply:
Okay.